esphome/components/spi/__init__.py

```diff
@@ -96,11 +93,7 @@
 
 
 def get_target_platform():
-    return (
-        CORE.data[KEY_CORE][KEY_TARGET_PLATFORM]
-        if KEY_TARGET_PLATFORM in CORE.data[KEY_CORE]
-        else ""
-    )
+    return CORE.data[KEY_CORE][KEY_TARGET_PLATFORM]
```

Comment:
There are circumstances (e.g. running script/build_language_schema.py) when there is no defined target platform, so the key can't be assumed to be present.
```suggestion
    return CORE.data[KEY_CORE].get(KEY_TARGET_PLATFORM, "")
```

Reply:
Good point. The value should be set inside build_language_schema.py similar to https://github.com/tomaszduda23/esphome/blob/14050411cd275ee40332090040adb6f98907bf8f/script/build_codeowners.py#L42. Otherwise there is no way to load components in correct order.

Reply:
Updated build_language_schema.py. list-components.py and build_codeowners.py already have it.